ablySpec/RealtimeClientConnection.swift

```diff
@@ -765,22 +765,20 @@ class RealtimeClientConnection: QuickSpec {
                         let transport = client.transport as! TestProxyTransport
                         transport.actionsIgnored += [.Ack, .Nack]
 
-                        waitUntil(timeout: testTimeout + options.disconnectedRetryTimeout) { done in
-                            channel.on { errorInfo in
-                                if channel.state == .Attached {
-                                    channel.publish(nil, data: "message", cb: { errorInfo in
-                                        expect(errorInfo).toNot(beNil())
-                                        done()
-                                    })
-                                    // Wait until the message is pushed to Ably first
-                                    delay(1.0) {
-                                        client.simulateLostConnection()
-                                        expect(client.connection.state).toEventually(equal(ARTRealtimeConnectionState.Connecting), timeout: options.disconnectedRetryTimeout)
-                                    }
-                                }
-                            }
-                            channel.attach()
+                        channel.attach()
+                        expect(channel.state).toEventually(equal(ARTRealtimeChannelState.Attached), timeout: testTimeout)
+
+                        var gotPublishedCallback = false
+                        channel.publish(nil, data: "message", cb: { errorInfo in
+                            expect(errorInfo).toNot(beNil())
+                            gotPublishedCallback = true
+                        })
+
+                        // Wait until the message is pushed to Ably first
+                        delay(1.0) {
+                            client.simulateLostConnection()
                         }
+                        expect(gotPublishedCallback).toEventually(beTrue(), timeout: testTimeout)
```

Comment:
I am not sure this is quite right. It's not clear that the state being lost has triggered the failure as opposed to just a timeout.  What we need to check is that as soon as the new connection is made, and the connection state is lost because the connection ID has changed, then immediately all messages should be treated as failed to be published.